File: whatscooking/src/main/java/com/rll/whatscooking/domain/Comments.java
package com.rll.whatscooking.domain;

import jakarta.persistence.Entity;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import jakarta.persistence.UniqueConstraint;
import lombok.Data;

@Entity
@Data
@Table(name = "comments", uniqueConstraints = {
    @UniqueConstraint(columnNames = {"user_id", "recipe_id"})})
public class Comments {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private int commentId;
    private String comment;

    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "user_id")
    private User user;

    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "recipe_id")
    private Recipe recipe;
}

File: Client/src/app/Models/Comments.ts
import { Recipe } from './Recipe';
import { User } from './User';

export class Comments {
  commentId?: number;
  comment?: string;
  user?: User;
  recipe?: Recipe;

  constructor(
    commentId?: number,
    comment?: string,
    user?: User,
    recipe?: Recipe
  ) {
    this.commentId = commentId;
    this.comment = comment;
    this.user = user;
    this.recipe = recipe;
  }
}
